201-openshift-origin-rhel/azuredeploy.json

```diff
@@ -0,0 +1,743 @@
+        "Standard_D12_v2",
+        "Standard_D13_v2",
+        "Standard_D14_v2",
+        "Standard_G1",
+        "Standard_G2",
+        "Standard_G3",
+        "Standard_G4",
+        "Standard_G5"
+      ],
+      "metadata": {
+        "description": "OpenShift Node VM(s) size"
+      }
+    },
+    "openshiftMasterHostname": {
+      "type": "string",
+      "minLength": 1,
+      "metadata": {
+        "description": "OpenShift Master hostname"
+      }
+    },
+    "openshiftMasterPublicIpDnsName": {
+      "type": "string",
+      "minLength": 1,
+      "metadata": {
+        "description": "OpenShift Master VM public IP DNS name"
+      }
+    },
+    "nodePrefix": {
+      "type": "string",
+      "minLength": 1,
+      "metadata": {
+        "description": "OpenShift Node VM prefix"
+      }
+    },
+    "nodeInstanceCount": {
+      "type": "int",
+      "minValue": 1,
+	  "allowedValues": [1,2,3,4,5,6,7,8,9,10],
+      "metadata": {
+        "description": "Number of OpenShift nodes"
+      }
+    },
+    "adminUsername": {
+      "type": "string",
+      "minLength": 1,
+      "metadata": {
+        "description": "Administrator username on all VMs"
+      }
+    },
+    "sshPublicKey": {
+      "type": "securestring",
+      "metadata": {
+        "description": "SSH public key on all VMs"
+      }
+    },
+    "sshPrivateKey": {
+      "type": "securestring",
+      "minLength": 1,
+      "metadata": {
+        "description": "SSH private key for deploying OpenShift"
+      }
+    },
+    "keyVaultName": {
+      "type": "string",
+      "minLength": 1,
+      "metadata": {
+        "description": "Name of Key Vault"
+      }
+    },
+    "azureAdUserObjectId": {
+      "type": "string",
+      "minLength": 1,
+      "metadata": {
+        "description": "Azure AD User Object ID for user with permissions to create Key Vault resource"
+      }
+    },
+    "azureAdAppName": {
+      "type": "string",
+      "metadata": {
+        "description": "Azure AD Application name"
+      }
+    },
+    "azureAdClientId": {
+      "type": "string",
+      "minLength": 1,
+      "metadata": {
+        "description": "Azure AD Application Client ID"
+      }
+    },
+    "azureAdClientSecret": {
+      "type": "securestring",
+      "minLength": 1,
+      "metadata": {
+        "description": "Azure AD Application Client Secret"
+      }
+    },
+    "azureAdTenantId": {
+      "type": "string",
+      "minLength": 1,
+      "metadata": {
+        "description": "Azure AD Tenant ID"
+      }
+    },
+    "azureAdLogoutRedirectUri": {
+      "type": "string",
+      "minLength": 1,
+      "metadata": {
+        "description": "Azure AD Application logout redirect URI"
+      }
+    }
+  },
+  "variables": {
+    "location": "[resourceGroup().location]",
+    "storageAccountType": "Standard_LRS",
+    "namingInfix": "[toLower(parameters('nodePrefix'))]",
+    "newStorageAccountSuffix": "[concat(variables('namingInfix'), 'sa')]",
```

Comment:
Using a user entered value "nodePrefix" here is going to cause failures when a user enters a special char. See #944  for a better way